jsapp/js/account/plan.component.tsx

```diff
@@ -0,0 +1,339 @@
+import React, {useEffect, useReducer, useState} from 'react';
+import styles from './plan.module.scss';
+import {getOrganization, getProducts, getSubscription, Product, Organization, BaseSubscription} from './stripe.api';
+import {handleApiFail} from 'js/utils';
+import Icon from '../components/common/icon';
+import {ROOT_URL} from 'js/constants';
+
+interface PlanState {
+  isLoading: boolean;
+  subscribedProduct: BaseSubscription;
+  intervalFilter: string;
+  filterToggle: boolean;
+  products: Product[];
+  organization: null | Organization;
+}
+
+// An interface for our action
+interface dataUpdates {
+  type: string;
+  prodData: any;
+}
+
+const initialState = {
+  isLoading: true,
+  subscribedProduct: [],
+  intervalFilter: 'year',
+  filterToggle: false,
+  products: [],
+  organization: null,
+};
+
+function planReducer(state: PlanState, action: dataUpdates){
+  switch (action.type){
+    case 'initialProd':
+      return {...state, products: action.prodData};
+    case 'initialOrg':
+      return {...state, organization: action.prodData};
+    case 'initialSub':
+      return {...state, subscribedProduct: action.prodData};
+    case 'month':
+      return {...state, intervalFilter: 'month', filterToggle: !state.filterToggle};
+    case 'year':
+      return {...state, intervalFilter: 'year', filterToggle: !state.filterToggle};
+    default:
+      return state;
+  }
+}
+
+export default function Plan() {
+  const [state, dispatch] = useReducer(planReducer, initialState);
+  const [expandComparison, setExpandComparison] = useState(false);
+  const [buttonsDisabled, setButtonDisabled] = useState(false);
+
+  useEffect(() => {
+    getProducts().then((data) => {
+      dispatch({
+        type: 'initialProd', prodData: data.results,
+      });
+    });
+
+    getOrganization().then((data) => {
+      console.log('org', data);
+
+      dispatch({
+        type: 'initialOrg', prodData: data.results[0],
+      });
+    });
+
+    getSubscription().then((data) => {
+      console.log('sub', data);
+      dispatch({
+        type: 'initialSub', prodData: data.results,
+      });
+    });
+  }, []);
+
+  // Filter prices based on plan interval
+  const filterPrices = () => {
+    if (state.products.length > 0) {
+      const filterAmount = state.products.map((product: any) => {
+        const filteredPrices = product.prices.filter((price: any) => {
+          const interval = price.human_readable_price.split('/')[1];
+          return interval === state.intervalFilter || price.unit_amount === 0;
+        });
+        return {
+          ...product,
+          prices: filteredPrices.length ? filteredPrices[0] : null,
+        };
+      });
+      return filterAmount.filter((product: any) => product.prices);
+    }
+  };
+
+  const isSubscribedProduct = (product: any) => {
+    return product.name === state.subscribedProduct?.name;
+  };
```

Comment:
It looks like I didn't write this function correctly, which is why the "Your Plan" tab isn't showing up on the frontend.

```suggestion
  const isSubscribedProduct = (product: Product) => {
    if( product.prices.unit_amount === 0 && !state.subscribedProduct?.length ) {
      return true;
    }
    return product.name === state.subscribedProduct?.name;  };
  };
```